packages/vaex-contrib/vaex/contrib/io/gbq.py

```diff
@@ -0,0 +1,194 @@
+    :param str export: Pass an filename or path to download the table as an Apache Arrow file, and leverage memory mapping. If `None` the DataFrame is in memory.
+    :param str client_project: The ID of the project that executes the query. Will be passed when creating a job. If `None`, it will be set with the same value as `project`.
+    :param credentials: The authorization credentials to attach to requests. See google.auth.credentials.Credentials for more details.
+    :rtype: DataFrame
+
+    Example:
+
+    >>> import os
+    >>> os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '../path/to/project_access_key.json'
+    >>> from vaex.contrib.io.gbq import download_google_bigquery_table
+
+    >>> client_project = 'my_project_id'
+    >>> project = 'bigquery-public-data'
+    >>> dataset = 'ml_datasets'
+    >>> table = 'iris'
+    >>> columns = ['species', 'sepal_width', 'petal_width']
+    >>> conditions = 'species = "virginica"'
+    >>> df = download_google_bigquery_table(project=project,
+                                            dataset=dataset,
+                                            table=table,
+                                            columns=columns,
+                                            condition=conditions,
+                                            client_project=client_project)
+    >>> df.head(3)
+    #    sepal_width    petal_width  species
+    0            2.5            1.7  virginica
+    1            2.5            2    virginica
+    2            2.2            1.5  virginica
+    >>>
+
+    '''
+    # Instantiate the table path and the reading session
+    bq_table = f'projects/{project}/datasets/{dataset}/tables/{table}'
+    req_sess = google.cloud.bigquery_storage.types.ReadSession(table=bq_table, data_format=google.cloud.bigquery_storage.types.DataFormat.ARROW)
+
+    # Read options
+    req_sess.read_options.selected_fields = columns
+    req_sess.read_options.row_restriction = condition
+
+    # Instantiate the reading client
+    client = google.cloud.bigquery_storage.BigQueryReadClient(credentials=credentials)
+
+    parent = f'projects/{client_project or project}'
+    session = client.create_read_session(parent=parent, read_session=req_sess, max_stream_count=1)
+    reader = client.read_rows(session.streams[0].name)
+
+    if export is None:
+        arrow_table = reader.to_arrow(session)
+        return vaex.from_arrow_table(arrow_table)
+
+    else:
+        # We need to get the schema first - Get one RecordsBatch manually to get the schema
+        # Get the pages iterator
+        pages = reader.rows(session).pages
+        # Get the first batch
+        first_batch = pages.__next__().to_arrow()
+        # Get the schema
+        schema = first_batch.schema
+
+        # This does the writing - streams the batches to disk!
+        with vaex.file.open(path=export, mode='wb') as sink:
```

Comment:
Missing fs_options :)

Reply:
Why would you want to use fs/fs_options here? Would one want to download the bq data to an external file system? Using it would require to download it again locally..  Maybe there is a usecase for it.. i'd not complicate it at the moment and keep this in mind in case somebody needs/asks for it..